src/field/berry.c

```diff
@@ -1099,91 +1126,96 @@ void ClearBerryTrees(void)
         tree->watered2 = 0;
         tree->watered3 = 0;
         tree->watered4 = 0;
         tree->berryYield = 0;
-        tree->stage = 2;
-        if (++tree->regrowthCount == 10)
+        tree->stage = BERRY_STAGE_SPROUTED;
+        if (++tree->regrowthCount == BERRY_REGROW_LIMIT)
             *tree = gBlankBerryTree;
         break;
     }
     return TRUE;
 }
 
-void BerryTreeTimeUpdate(s32 minutes)
+// totalMinutes is how long its been since the last berry tree update.
+void BerryTreeTimeUpdate(s32 totalMinutes)
 {
     int i;
     struct BerryTree *tree;
 
-    for (i = 0; i < (u8)ARRAY_COUNT(gSaveBlock1.berryTrees); i++)
+    for (i = 0; i < MAX_BERRY_TREES; i++)
     {
         tree = &gSaveBlock1.berryTrees[i];
 
-        if (tree->berry && tree->stage && !tree->growthSparkle)
+        if (tree->berry != BERRY_NONE && tree->stage != BERRY_STAGE_NO_BERRY && tree->growthSparkle == FALSE)
         {
-            if (minutes >= GetStageDurationByBerryType(tree->berry) * 71)
+            // the player has waited too long to water the berry. Reset the tree. This is because if the berry state is not in the unwatered state, the tree will grow anyway despite this check, which means BerryTreeGrow will handle the regrow process for this, removing the need for this check. This only handles the unwatered soil state.
+            if (totalMinutes >= GetStageDurationByBerryType(tree->berry) * 71)
             {
                 *tree = gBlankBerryTree;
             }
             else
             {
-                s32 time = minutes;
+                // not sure why Game Freak thought a temporary variable was necessary. because time is altered below, perhaps they thought it was unsafe to change it, even though that is not how passed arguments behave.
+                s32 time = totalMinutes;
 
                 while (time != 0)
                 {
                     if (tree->minutesUntilNextStage > time)
                     {
-                        tree->minutesUntilNextStage -= time;
+                        tree->minutesUntilNextStage -= time; // its been X minutes since the last berry update, so update minutesUntilNextStage appropriately to match the time offset that has passed since the update.
                         break;
                     }
+                    // perform the subtraction the other way around to get the number of minutes since the inferred stage update that occured, since minutesUntilNextStage is <= time. we may need this variable to simulate multiple berry cycles in the while loop.
                     time -= tree->minutesUntilNextStage;
-                    tree->minutesUntilNextStage = GetStageDurationByBerryType(tree->berry);
-                    if (!BerryTreeGrow(tree))
+                    tree->minutesUntilNextStage = GetStageDurationByBerryType(tree->berry); // since the tree was inferred to update, set the new minutesUntilNextStage.
+                    if (BerryTreeGrow(tree) == FALSE)
```

Comment:
Why is this changed to be explicit?

Reply:
It is a boolean check. I would rather code be explicit than someone easily glance over it and make mistakes because a tiny, hard to see "!".